### DISCUSSION TITLE
Passing large strings from C# to C++ via `Actor.Invoke()`

### DISCUSSION THREAD

**Discussion by tonetechnician** [nxrighthere/UnrealCLR#159] [Help] [answered]
Hey there again! 

I'm trying to figure a good way to pass structs from C# to Cpp. One approach I've been using is to serialize my C# object to a JSON string before passing it to cpp via Actor.Invoke. Once in Cpp I can deserialize using ue's regular Json deserialization methods. This works really well for smaller structures but when trying to pass a larger data structure to cpp I get a UnrealFramework.StackOverflow error. 

Whilst this is not ideal, I've tried to avoid the StackOverflow by sending the serialized string in chunks but this turns out to open a can of worms as I start getting other errors when trying to interact with the actor that I've invoked my functions - `System.InvalidOperationException: Operation is not valid due to the current state of the object.`

This seems quite intermittent so I'm worried about using this and would opt for a cleaner way of passing the data if there is one.

The string I'm trying to pass has a char length of roughly 73383 characters. 

Does anyone think there's a better way to pass a structure like this across? Would be super appreciative if anyone has ideas of how to allow this

**Comment by tonetechnician** [comment, marked as answer]
Sorry, I've just noticed this is actually an issue in my own code. Passing the my large string is actually working well.

**Reply by nxrighthere** [reply]
There are no limitations for strings length on the plugin side for invocation of Unreal functions.

**Reply by tonetechnician** [reply]
Indeed! Thanks for confirmation.

Apologies, had the stack overflow error in my code base and it was displaying an error as coming from UnrealFramework.StackOverflow.

Thanks again for all your input @nxrighthere ! Seriously great work you're doing with this integration. Looking forward to trying in UE5 :)
